go/testing/action.go

```diff
@@ -69,11 +73,16 @@ func RunTestCase(graph *Graph, driver Driver, test *pf.TestSpec) (scope string,
 
 		// Advance time to unblock the next PendingStat.
 		if nextReady != -1 {
-			if err := driver.Advance(nextReady); err != nil {
+			err := driver.Advance(nextReady)
+			if err == ErrAdvanceDisabled {
+				log.Warnf("time advance disabled with action in %s.", nextReady)
```

Comment:
nit: use structured logging, like:

```
log.WithFields(log.Fields{"delay": nextReady}).Warn("the message")
```

I suspect this warning is going to be pretty confusing to users. We should plumb through more context for the user about _why_ there's a delay. Can we get the name of the catalog task ? Probably means having PopReadyStats pass back more info.

Also note that `nextReady` is an absolute duration that's relative to the test start time, so converting to a deadline might be more useful (particularly because there can be multiple delays).

Reply:
I tried to make it more obvious. The time being returned is the actual delta time, so zero if there's something to do otherwise some time.Duration. I just mad the message, `sleeping until next task is ready.` If I'm there, I wanna know how long I am going to be waiting.